doc/lua_api.txt

```diff
@@ -2268,7 +2269,101 @@ Elements
 **Note**: do _not_ use a element name starting with `key_`; those names are
 reserved to pass key press events to formspec!
 
+### `style[<name>;<prop1>;<prop2>;...]`
 
+Set the style for the named element `name`.
+Note: this **must** be before the element is defined.
+
+See [Styling Formspecs].
+
+
+### `style_type[<type>;<prop1>;<prop2>;...]`
+
+Sets the style for all elements of type `type` which appear after this element.
+
+See [Styling Formspecs].
+
+
+
+Styling Formspecs
+-----------------
+
+Formspec elements can be themed using the style elements:
+
+    style[<name>;<prop1>;<prop2>;...]
+    style_type[<type>;<prop1>;<prop2>;...]
+
+Where a prop is:
+
+    property_name=property_value
+
+For example:
+
+    style_type[button;bgcolor=#006699]
+    style[world_delete;bgcolor=red;textcolor=yellow]
+    button[4,3.95;2.6,1;world_delete;Delete]
+
+Setting a property to nothing will reset it to the default value. For example:
+
+    style_type[button;bgimg=button.png;bgimg_pressed=button_pressed.png;border=false]
+    style[btn_exit;bgimg=;bgimg_pressed=;border=;bgcolor=red]
+
+
+### Supported Element Types
+
+Some types may inherit styles from parent types.
+
+* button
+* button_exit, inherits from button
+* checkbox
+* scrollbar
+* table
+* textlist
+* dropdown
+* field
+* pwdfield, inherits from field
+* textarea
+* label
+* vertlabel, inherits from field
+* image_button
+* item_image_button, inherits from image_button
+* tabheader
+
+
+### Valid Properties
+
+* button, button_exit
+    * bgcolor - color, sets button tint
+    * textcolor - color, default white
+    * border - boolean, draw border. Set to false to hide the bevelled button pane.  Default true.
+    * noclip - boolean, set to true to show things over the element bounds. Default false.
+    * bgimg - standard image. Defaults to none.
+    * bgimg_pressed - image when pressed. Defaults to bgimg when not provided.
+    * alpha - boolean, whether to draw alpha in bgimg. Default true.
+* checkbox
+    * noclip - boolean, set to true to show things over the element bounds. Default false.
+* scrollbar
+    * noclip - boolean, set to true to show things over the element bounds. Default false.
+* table, textlist
+    * noclip - boolean, set to true to show things over the element bounds. Default false.
+* dropdown
+    * noclip - boolean, set to true to show things over the element bounds. Default false.
+* field, pwdfield, textarea
+    * noclip - boolean, set to true to show things over the element bounds. Default false.
```

Comment:
What use does `noclip` have in this case? The text is always inside the element bounds.

Reply:
The element itself can be outside of the formspec window (made with `size`).

I can't really imagine any usecase for this. Why would anyone want to add an element outside of the `size` borders? If they want to have transparent background, they can use `bgcolor`.

Reply:
This can be useful.  For instance, if a person makes a formspec designer, the fields for the coordinates of each element would have to be outside the form to not conflict with the things inside it.

This should probably be clarified to say outside the formspec window, not the element bounds.

Reply:
![image](https://user-images.githubusercontent.com/2122943/59283681-da613c80-8c62-11e9-9745-03a2254d9c23.png)